adapters/celestia/src/verifier/address.rs

```diff
@@ -1,64 +1,116 @@
 use std::fmt::{Display, Formatter};
 use std::str::FromStr;
 
+use bech32::WriteBase32;
 use borsh::{BorshDeserialize, BorshSerialize};
 use serde::{Deserialize, Serialize};
 use sov_rollup_interface::AddressTrait;
 
+/// Human Readable Part: "celestia" for Celestia network
 const HRP: &str = "celestia";
 
+/// Representation of the address in the Celestia network
 #[derive(Debug, PartialEq, Clone, Eq, Serialize, Deserialize, BorshDeserialize, BorshSerialize)]
-// Raw ASCII bytes, including HRP
-// TODO: https://github.com/Sovereign-Labs/sovereign-sdk/issues/469
-pub struct CelestiaAddress(Vec<u8>);
+pub struct CelestiaAddress {
+    // Storing vector of u5 each as u8
+    data: Vec<u8>,
+    // indication if it is Bech32m or not
+    m: bool,
```

Comment:
Does Celestia dynamically select between the standard and `m` variants of bech32? I thought it always used the standard variant

Reply:
Is it possible to make it two variant enum instead of having a flag?

Reply:
@preston-evans98 , you are correct, currently celestia and cosmos sdk use `bech32`. When(if) they upgrade, we will do as well. But I believe that this upgrade might be impossible, as string representation of address will change.